extensions/amp-form/0.1/amp-form.js

```diff
@@ -86,20 +90,24 @@ export class AmpForm {
 
   /** @private */
   installSubmitHandler_() {
-    this.form_.addEventListener('submit', e => this.handleSubmit_(e));
+    this.form_.addEventListener('submit', e => this.handleSubmit_(e), true);
   }
 
   /**
+   * Note on stopImmediatePropagation usage here, it is important to emulate native
+   * browser submit event blocking. Otherwise any other submit listeners would get the
+   * event.
+   *
+   * For example, action service shouldn't trigger 'submit' event if form is actually
+   * invalid. stopImmediatePropagation allows us to make sure we don't trigger it
```

Comment:
For the XHR it might actually make sense to set a property on the event.

However, what I meant is that when `stopImmediatePropagation` is important over `stopPropagation` you need to be super careful about order. But the way you currently set up handlers registration order does not make a difference.